components/common/src/package_graph.rs

```diff
@@ -12,48 +12,70 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use bimap::BiMap;
 use error::Result;
+use hcore::fs as hfs;
+use hcore::package;
 use hcore::package::ident::PackageIdent;
 use hcore::package::PackageInstall;
 use petgraph;
 use petgraph::graph::NodeIndex;
 use petgraph::stable_graph::StableGraph;
-use petgraph::visit::Bfs;
-use std::collections::HashMap;
+use petgraph::visit::{Bfs, Reversed};
+use std::path::Path;
 
 pub struct PackageGraph {
-    nodes: HashMap<PackageIdent, NodeIndex>,
+    nodes: BiMap<PackageIdent, NodeIndex>,
     graph: StableGraph<PackageIdent, usize, petgraph::Directed>,
 }
 
 impl PackageGraph {
-    pub fn new() -> Self {
+    fn empty() -> Self {
         PackageGraph {
-            nodes: HashMap::<PackageIdent, NodeIndex>::new(),
-            graph: StableGraph::<PackageIdent, usize>::new(),
+            nodes: BiMap::new(),
+            graph: StableGraph::new(),
         }
     }
 
-    // Given a list of packages build a directed package graph
-    // using the package dependencies
-    pub fn build(&mut self, packages: &Vec<PackageInstall>) -> Result<(usize, usize)> {
```

Comment:
Sorry I'm late to the party... I am a bit concerned that we have two separate implementations of PackageGraph now, one in habitat and the other in builder, and they are diverging APIs.

Reply:
That's probably worth resolving and pulling into core.

Reply:
Yep, I've opened https://github.com/habitat-sh/core/issues/94 to cover that work as a separate issue.